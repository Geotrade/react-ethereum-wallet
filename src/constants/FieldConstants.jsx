export const AccountPageHeader = [
  {text: "Accounts", strong: true},
  {text: " ", strong: false},
  {text: "Overiew", strong: false},
];

export const ContractPageHeader = [
  {text: "Contracts", strong: true},
];

export const SendPageHeader = [
  {text: "Send", strong: true},
  {text: " ", strong: false},
  {text: "funds", strong: false},
];

export const DefaultAccountList = [
  {
    title: 'Accounts',
    redirect: false,
    buttonClass: 'wallet-box create add-account',
    buttonDescription: 'ADD ACCOUNT',
    contractDescription:
      "Accounts are password protected keys that can hold Ether and Ethereum-based tokens. They can control contracts, but can't display incoming transactions."
  },
  {
    title: 'Wallet Contracts',
    redirect: true,
    link: 'account/new',
    buttonClass: 'wallet-box create ',
    buttonDescription: 'ADD WALLET CONTRACt',
    contractDescription:
      'These contracts are stored on the blockchain and can hold and secure Ether. They can have multiple accounts as owners and keep a full log of all transactions.'
  }
];


export const DefaultContractList = [
  {
    title: '',
    redirect: true,
    link: 'deploy-contract',
    buttonClass: 'wallet-box create add-contract',
    buttonDescription: 'DEPLOY NEW CONTRACT',
    contractDescription: ''
  },
  {
    title: 'Custom Contracts',
    redirect: false,
    link: 'deploy-contract',
    buttonClass: 'wallet-box create add-contract',
    buttonDescription: 'WATCH CONTRACT',
    contractDescription:
      'To watch and interact with a contract already deployed on the blockchain, you need to know its address and the description of its interface in JSON format.'
  },
  {
    title: 'Custom Tokens',
    redirect: false,
    link: 'deploy-contract',
    buttonClass: 'wallet-box create add-token',
    buttonDescription: 'DEPLOY NEW CONTRACT',
    contractDescription:
      'Tokens are currencies and other fungibles built on the Ethereum platform. In order for accounts to watch for tokens and send them, you have to add their address to this list. You can create your own token by simply modifying this example of a custom token contract or learning more about Ethereum Tokens.'
  }
];