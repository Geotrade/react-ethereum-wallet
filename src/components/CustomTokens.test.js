import React from 'react';
import renderer from 'react-test-renderer';
import { ContractSectionList } from '../../constants/FieldConstants.js';

describe('custom contracts html', () => {
  const CC = ContractSectionList.CustomContracts;
});
