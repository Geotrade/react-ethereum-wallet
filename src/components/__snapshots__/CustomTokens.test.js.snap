// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom contracts html renders auto scan 1`] = `
Array [
  <button
    className="wallet-box create token-auto-scan"
    title="Automatically scan for balances of popular tokens on your accounts."
>
    <div
        className="account-pattern"
    >
        <div
            className="icon icon-target"
        />
    </div>
    <h3>
        Auto-Scan
    </h3>
</button>,
  <div
    className="dapp-clear-fix"
/>,
]
`;

exports[`custom contracts html renders token box 1`] = `
<button
  className="wallet-box create add-token"
  onClick={
    Object {
      "modals": Object {
        "displayWatchToken": true,
      },
    }
  }
  style={
    Object {
      "float": "left",
    }
  }
>
  <div
    className="account-pattern"
  >
    +
  </div>
  <h3>
    WATCH CUSTOM TOKEN
  </h3>
</button>
`;

exports[`custom contracts html renders token description 1`] = `
Array [
  <h2>
    Custom Tokens
</h2>,
  <p>
    Tokens are currencies and other fungibles built on the Ethereum platform. In order for accounts to watch for tokens and send them, you have to add their address to this list. You can create your own token by simply modifying this example of a custom token contract or learning more about Ethereum Tokens.
</p>,
  <div
    className="dapp-clear-fix"
/>,
]
`;
